--- a/src/plugins/meson.build	2022-09-07 16:04:47.034037109 +0200
+++ b/src/plugins/meson.build	2022-09-07 16:04:47.036037223 +0200
@@ -69,7 +69,7 @@
   endif
 endif
 
-pylint_flags = ['-d', 'C0111', '-d', 'W0511', '-d', 'F0401', '-d', 'C0326' ]
+pylint_flags = ['-d', 'C0111', '-d', 'W0511', '-d', 'F0401', '-d', 'C0326', '--load-plugins=pylint.extensions.no_self_use' ]
 
 # FIXME: Move custom_target to i18n.merge_file when gettext acquires plugin support.
 #        http://lists.gnu.org/archive/html/bug-gettext/2017-06/msg00001.html
 
--- a/src/plugins/pythonconsole/console.py	2022-09-07 16:09:54.982742285 +0200
+++ b/src/plugins/pythonconsole/console.py	2022-09-07 16:10:03.800263422 +0200
@@ -316,20 +316,20 @@
         pass
     def fileno(self):
         return self.file_no
-    def isatty(self): # pylint: disable=R0201
+    def isatty(self): # pylint: disable=no-self-use
         return 0
-    def read(self, _): # pylint: disable=R0201
+    def read(self, _): # pylint: disable=no-self-use
         return ''
-    def readline(self): # pylint: disable=R0201
+    def readline(self): # pylint: disable=no-self-use
         return ''
-    def readlines(self): # pylint: disable=R0201
+    def readlines(self): # pylint: disable=no-self-use
         return []
     def write(self, seg):
         self.console.write(seg, self.tag)
     def writelines(self, lines):
         self.console.write(lines, self.tag)
-    def seek(self, _): # pylint: disable=R0201
+    def seek(self, _): # pylint: disable=no-self-use
         raise IOError((29, 'Illegal seek'))
-    def tell(self): # pylint: disable=R0201
+    def tell(self): # pylint: disable=no-self-use
         raise IOError((29, 'Illegal seek'))
     truncate = tell
