From ee97e2bf2a57126f2ef29e1dfc2cf29413a11e6c Mon Sep 17 00:00:00 2001
From: Fabian Bornschein <1885-fabiscafe@users.noreply.gitlab.gnome.org>
Date: Sun, 18 Sep 2022 11:04:35 +0000
Subject: [PATCH] Fix build with shumate 1.0.0

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 375965f7..5b1a0ee4 100644
--- a/meson.build
+++ b/meson.build
@@ -42,7 +42,7 @@ gweather = dependency('gweather4', version: '>= 3.90.0')
 geocodeglib = dependency('geocode-glib-2.0', version: '>= 3.15.2')
 
 libmaps_deps = [
-	dependency('shumate-1.0', version: '>= 1.0.0.beta'),
+	dependency('shumate-1.0', version: '>= 1.0.0'),
 	dependency('libxml-2.0'),
 	dependency('rest-1.0')
 ]
-- 
GitLab

