From 3bad03e90fcc28f6e3f87f2c90df5984dbeb0791 Mon Sep 17 00:00:00 2001
From: Bobby Rong <rjl931189261@126.com>
Date: Wed, 3 Aug 2022 19:43:22 +0000
Subject: [PATCH] build: Fix building with -Werror=format-security

---
 src/gui/gtd-window.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/gui/gtd-window.c b/src/gui/gtd-window.c
index 783357d4..91323ff4 100644
--- a/src/gui/gtd-window.c
+++ b/src/gui/gtd-window.c
@@ -142,6 +142,7 @@ error_message_notification_secondary_action (GtdNotification *notification,
                                            NULL);
 
   adw_message_dialog_format_body (ADW_MESSAGE_DIALOG (dialog),
+                                  "%s",
                                   data->secondary_text);
 
   adw_message_dialog_add_response (ADW_MESSAGE_DIALOG (dialog),
-- 
GitLab

